test/Test/Data/Mutable/Array.hs

```diff
@@ -6,31 +6,14 @@
 -- |
 -- Tests for mutable arrays.
 --
--- The API is basically 4 mutators or constructors: alloc, write, resize,
--- resizeSeed and 2 accessors: length, read.
+-- See the testing framework explained in Test.Data.Mutable.Set.
 --
--- We test arrays behave properly with properties that axiomatically define
--- arrays.  Basically, using these properties, the behavior of any use of
--- arrays should be fully specified without having to look at the
--- implementation.
+-- The combination of axioms and homomorphisms provided fully specify
+-- the behavior of arrays.
```

Comment:
I think it's a bit of a misnomer to call these things homorphisms (there is a sense in which they are, but I don't think it's really a meaningful sense). What we are doing is testing operations with respect to their semantics (which happens to be given as a reference implementation).

Reply:
I think they are mostly like homomorphisms: we carry through an operation `f (fromList l) = fromList (f' l)` and have a correspondence between `f` and `f'` subject to the map `fromList`. As a result of this, because we have a semantics for `f'`, we specify `f` because any computation done with an `f` can be translated to one done with `f'`. This will be explained in the tests for `Set`.